grakn-engine/src/main/java/ai/grakn/engine/backgroundtasks/distributed/TaskFailover.java

```diff
@@ -144,8 +145,14 @@ private void reQueue(CuratorFramework client, String engineID) throws Exception
             // Mark task as SCHEDULED again
             TaskState taskState = stateStorage.getState(id);
 
-            stateStorage.updateState(taskState.status(SCHEDULED));
-            producer.send(new ProducerRecord<>(WORK_QUEUE_TOPIC, id, taskState.configuration().toString()));
+            if(taskState.status() == RUNNING) {
+                LOG.debug(format("Engine [%s] stopped, task [%s] requeued", engineID, taskState.getId()));
+                stateStorage.updateState(taskState.status(SCHEDULED));
+                producer.send(new ProducerRecord<>(WORK_QUEUE_TOPIC, id, taskState.configuration().toString()));
+            } else {
+                LOG.debug(format("Engine [%s] stopped, task [%s] not restarted because state [%s]"
+                        , engineID, taskState.getId(), taskState.status()));
+            }
         }
     }
 
```

Comment:
That pair of operations in the reQueue method, "set state to schedule, then use producer to put in queue", is essentially the "schedule" operation. It's a how  a task is scheduled to run, correct? So ideally there should be a method that does precisely that - implement all the steps necessary to schedule a task. Not a big deal because those are two lines, but it's good to have all major steps of the workflow represented by distinct methods.